thrift/src/main/java/com/linecorp/armeria/server/thrift/THttpService.java

```diff
@@ -352,20 +353,20 @@ public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exc
         final HttpResponse res = HttpResponse.from(responseFuture);
         ctx.logBuilder().serializationFormat(serializationFormat);
         ctx.logBuilder().deferRequestContent();
-        req.aggregateWithPooledObjects(ctx.eventLoop(), ctx.alloc()).handle((aReq, cause) -> {
-            if (cause != null) {
-                final HttpResponse errorRes;
-                if (ctx.config().verboseResponses()) {
-                    errorRes = HttpResponse.of(HttpStatus.INTERNAL_SERVER_ERROR,
-                                               MediaType.PLAIN_TEXT_UTF_8,
-                                               Exceptions.traceText(cause));
-                } else {
-                    errorRes = HttpResponse.of(HttpStatus.INTERNAL_SERVER_ERROR);
-                }
-                responseFuture.complete(errorRes);
-                return null;
-            }
-
+        PooledHttpRequest.of(req).aggregateWithPooledObjects(
```

Comment:
This makes me wonder which is better - `req.toPooled()` vs. `PooledHttpRequest.of(req)`. Should we make our users mention the type explicitly just because it's an advanced API? 🤔

Reply:
Yeah thought about this but lean towards `PooledHttpRequest.of` - a bit more verbose, but it's the only way to keep the concept of unsafe completely out of the unsafe packages. I think this is valuable.

Reply:
I see. No problem.